modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/RelationshipObjectFieldBusinessType.java

```diff
@@ -87,69 +91,114 @@ public Object getValue(
 			ObjectField objectField, long userId, Map<String, Object> values)
 		throws PortalException {
 
-		String objectRelationshipERCObjectFieldName =
-			ObjectFieldSettingUtil.getValue(
-				ObjectFieldSettingConstants.
-					NAME_OBJECT_RELATIONSHIP_ERC_OBJECT_FIELD_NAME,
-				objectField);
+		String objectRelationshipObjectFieldName = StringUtil.split(
+			objectField.getName(), CharPool.UNDERLINE
+		).get(
+			1
+		);
 
 		if (Objects.equals(
 				objectField.getRelationshipType(),
 				ObjectRelationshipConstants.TYPE_ONE_TO_MANY) &&
-			values.containsKey(objectRelationshipERCObjectFieldName)) {
+			values.containsKey(objectRelationshipObjectFieldName) &&
+			(values.get(objectRelationshipObjectFieldName) instanceof Map)) {
+
+			Map<String, Object> nestedField = (Map<String, Object>)values.get(
```

Comment:
I think we should investigate why you found necessary evaluating the nested fields. I'm currently into it but I comment it just to have it in our minds.